File: client/src/app/models/craft-list-params.ts
import { PaginationParams } from "./pagination-params";

export enum ArchiveFilterType
{
    All = 1,
    NotArchivedOnly = 2,
    ArchivedOnly = 3
}

export class CraftListParams extends PaginationParams{
    minPrice = 0.0;
    maxPrice = 9999.0;
    inStockOnly = false;
    archiveFilter = ArchiveFilterType.NotArchivedOnly;
}

File: API/Pagination/CraftListParams.cs
namespace API.Pagination;

public enum ArchiveFilterType
{
    All = 1,
    NotArchivedOnly = 2,
    ArchivedOnly = 3
}

public class CraftListParams : PaginationParams
{
    public float MaxPrice { get; set; } = float.MaxValue;
    public float MinPrice { get; set;} = 0.0f; 

    public bool InStockOnly { get; set; } = false;
    public ArchiveFilterType ArchiveFilter{ get; set; } = ArchiveFilterType.All;
}